File: 原創不分類題庫/Contents/d874/d874.cpp
#include<iostream>
#include<cmath>

using namespace std;

int main(void)
{
	int L;
	
	while(cin >> L)
	{
		int temp = (int)sqrt(L);
		int step = 2 * temp - 1;
		
		step += (L - temp * temp) / temp;
		
		if(L % temp)
			step++;
		
		cout << step << endl;
	}
	
	return 0;
}

File: 原創不分類題庫/Contents/d874/d874.py
import sys

for L in sys.stdin:
    L = int(L)
    temp = int(L ** 0.5)
    step = 2 * temp - 1

    step += (L - temp ** 2) // temp

    if L % temp != 0:
        step += 1

    print(step)
